### PR TITLE
added classes

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

**Review by andriylagun — APPROVED** [review]
Great gob!

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,61 @@
...
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (dmytro-hryhoruk) [inline pom.xml]: we don't use it yet

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,61 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
```
Comment (dmytro-hryhoruk) [inline pom.xml]:
```suggestion
</project>
```

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public Product update(Product product) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(product.getId())) {
+                return Storage.products.set(i, product);
+            }
+        }
+        throw new NoSuchElementException("Such product was not found");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: let's pass product id to exception message

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.products.removeIf(x -> x.getId().equals(id));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]:
```suggestion
        return Storage.products
          .removeIf(p -> p.getId().equals(id));
```

Comment (Martynice) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: added

File: src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java (hunk 3 above)
Comment (Martynice) [reply src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java]: added

File: pom.xml (hunk 2 above)
Comment (Martynice) [reply pom.xml]: removed

File: pom.xml (hunk 1 above)
Comment (Martynice) [reply pom.xml]: removed

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**